frontend/src/App.tsx

```diff
@@ -1,66 +1,109 @@
-
+  "0xa0f2b97f32300d7360b5632e682c1508b326da9a00fbe755646d3809aec90672";
+ 
 function App() {
-  const [counter, setCounter] = useState(0);
-  const [loading, setLoading] = useState(false);
-
+  const [connected, setConnected] = useState<boolean>(false);
+  const [account, setAccount] = useState<string>("");
+  const [counter, setCounter] = useState<number>(0);
+  const [loaded, setLoaded] = useState(false);
+ 
   useEffect(() => {
-    async function main() {
-      // Executes the counter function to query the current contract state
-      // the `.get()` is read-only, because of this it don't expand coins.
-      const { value } = await contract.functions.count().get();
-      setCounter(Number(value));
+    setTimeout(() => {
+      checkConnection();
+      setLoaded(true);
+    }, 200)
+    if (connected) getCount();
+  }, [connected])
+ 
+  async function connect() {
+    if (window.fuel) {
+      try {
+        await window.fuel.connect();
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      } catch (err) {
+        console.log("error connecting: ", err);
+      }
     }
-    main();
-  }, []);
-
+  }
+ 
+  async function checkConnection() {
+    if (window.fuel) {
+      const isConnected = await window.fuel.isConnected();
+      if (isConnected) {
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      }
+    }
+  }
+ 
+  async function getCount() {
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+      const { value } = await contract.functions.count().simulate();
+      setCounter(value.toNumber());
+    }
+  }
+ 
   async function increment() {
-    // a loading state
-    setLoading(true);
-    // Creates a transactions to call the increment function
-    // because it creates a TX and updates the contract state this requires the wallet to have enough coins to cover the costs and also to sign the Transaction
-    try {
-      await contract.functions.increment().txParams({ gasPrice: 1 }).call();
-      const { value } = await contract.functions.count().get();
-      setCounter(Number(value));
-    } finally {
-      setLoading(false);
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+      // Creates a transactions to call the increment function
+      // because it creates a TX and updates the contract state this requires the wallet to have enough coins to cover the costs and also to sign the Transaction
+      try {
+        await contract.functions.increment().txParams({ gasPrice: 1 }).call();
+        getCount();
+      } catch (err) {
+        console.log("error sending transaction...", err);
+      }
     }
   }
-  
+ 
+  if (!loaded) return null
```

Comment:
Should it return a div that says loading, or a spinner instead?

Reply:
Agree with the "Loading..." 
The spinner might be _"too nice to have"_ in the context of a quickstart.